### PR TITLE
US128553: Migrate to postgres2.0 and aws-redis for int tests

### PR DESCRIPTION
**Pull request by FrankGasparovic**

### CODE REVIEW

**Comment by FrankGasparovic** [comment]
Cloud titan int tests: https://predix1.jenkins.build.ge.com/job/Predix-Security/job/ACS/job/ACS-PullRequest/job/acs-integration-pullrequest/611/badge/icon

Metering failed as usual but also connectorIT failed. Not sure if maybe asset was acting up or one of my changes caused it.

File: acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java [hunk 1]
```diff
@@ -95,24 +94,31 @@ void pushApplication(final Map<String, String> additionalEnvironmentVariables) t
...
 
-        Map<String, String> environmentVariables = new HashMap<>(COMMON_ENVIRONMENT_VARIABLES);
-        environmentVariables.putAll(additionalEnvironmentVariables);
-        environmentVariables.values().removeIf(Objects::isNull);
-
-        CloudFoundryService postgresService = new CloudFoundryService("shared-nr",
-                AcsCloudFoundryUtilities.ACS_DB_SERVICE_INSTANCE_NAME, "postgres", Collections.emptyMap());
-        CloudFoundryService redisService = new CloudFoundryService("shared-vm",
-                AcsCloudFoundryUtilities.ACS_REDIS_SERVICE_INSTANCE_NAME, "redis", Collections.emptyMap());
+        CloudFoundryService postgresService = new CloudFoundryService("dedicated-20.1", this.acsDbServiceName,
```
Comment (anubhavi25) [inline acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java]: Use `dedicated-10.1` and downgrade the plans for `acs-rds-postgres-ci-{0..5}` to `dedicated-10.1` as well if they were provisioned with the higher plan.

File: acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java [hunk 2]
```diff
@@ -124,4 +130,13 @@ void pushApplication(final Map<String, String> additionalEnvironmentVariables) t
         System.setProperty("ASSET_ZONE_ID", assetZoneId);
         System.setProperty("ASSET_URL", assetUrl);
     }
+
+    public String getAcsRedisServiceName() {
```
Comment (anubhavi25) [inline acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java]: Can this function be made package-private?

File: acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java [hunk 3]
```diff
@@ -124,4 +130,13 @@ void pushApplication(final Map<String, String> additionalEnvironmentVariables) t
...
+        return acsRedisServiceName;
+    }
+
+
+    public String getAcsDbServiceName() {
```
Comment (anubhavi25) [inline acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java]: Can this function be made package-private?

File: acs-integration-tests/src/test/java/com/ge/predix/cloudfoundry/client/CloudFoundryApplicationHelper.java [hunk 4]
```diff
@@ -217,6 +226,22 @@ public void bindServiceInstances(final String applicationName, final List<String
                         .build()));
     }
 
+    public String getUnusedServiceInstance(final List<String> instanceNames) {
+
+        for (String instance : instanceNames) {
+            try {
+                if (this.cloudFoundryOperations.services().getInstance(GetServiceInstanceRequest.builder()
+                        .name(instance).build()).block().getApplications().size() == 0) {
```
Comment (anubhavi25) [inline acs-integration-tests/src/test/java/com/ge/predix/cloudfoundry/client/CloudFoundryApplicationHelper.java]: Use `isEmpty()`

File: acs-integration-tests/src/test/java/com/ge/predix/cloudfoundry/client/CloudFoundryApplicationHelper.java [hunk 5]
```diff
@@ -217,6 +226,22 @@ public void bindServiceInstances(final String applicationName, final List<String
...
+                    return instance;
+                }
+            } catch (RuntimeException e) {
+                LOGGER.info(String.format("Instance '%s' does not exist", instance), e);
```
Comment (anubhavi25) [inline acs-integration-tests/src/test/java/com/ge/predix/cloudfoundry/client/CloudFoundryApplicationHelper.java]: Use built-in formatter: `LOGGER.info("Instance '{}' does not exist", instance, e);`

**Review by anubhavi25 — CHANGES_REQUESTED** [review]
As discussed, we should also add logic to flush the bound cache and drop/create the bound database before commencing integration tests.

File: acs-integration-tests/src/test/java/com/ge/predix/cloudfoundry/client/CloudFoundryApplicationHelper.java (hunk 5 above)
Comment (FrankGasparovic) [reply acs-integration-tests/src/test/java/com/ge/predix/cloudfoundry/client/CloudFoundryApplicationHelper.java]: I don't think info has a signature with params and throwable 
https://www.slf4j.org/apidocs/org/slf4j/Logger.html

File: acs-integration-tests/src/test/java/com/ge/predix/cloudfoundry/client/CloudFoundryApplicationHelper.java (hunk 4 above)
Comment (FrankGasparovic) [reply acs-integration-tests/src/test/java/com/ge/predix/cloudfoundry/client/CloudFoundryApplicationHelper.java]: Fixed

File: acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java (hunk 3 above)
Comment (FrankGasparovic) [reply acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java]: Fixed

File: acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java (hunk 2 above)
Comment (FrankGasparovic) [reply acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java]: Fixed

File: acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java (hunk 1 above)
Comment (FrankGasparovic) [reply acs-integration-tests/src/test/java/com/ge/predix/acs/cloudfoundry/PushAcsApplication.java]: Fixed

**Review by anubhavi25 — APPROVED** [review]

**Comment by anubhavi25** [comment]
[Builds #616-617,619-620 Passing (`predix-cloud*` test failures are not related to this PR)](https://predix1.jenkins.build.ge.com/job/Predix-Security/job/ACS/job/ACS-PullRequest/job/acs-integration-pullrequest/)

**Comment by FrankGasparovic** [comment]
Looks good

**Pull request merged**